### ISSUE TITLE
Suggestion list

### ISSUE THREAD

**Issue by zBoringDrop** [antiAD/AntiAd#80]
Hi, there is my suggestion for this plugin:

- Move this project on spigot.
- Make a better system of detecting IPs (actual is very bad). So this new detect system will check EVERY words wrote in a text of a player. For Example: "Hi, join in Mc.servername.it!", your plugin will search if this ip works, and if work, the plugin will stop the text from its send. If the text is: "Hi, Join in Mc.serverthatnotexist.it" the plugin will see that this ip is not real and it doesn't stop the send of the text.
- Make a new config: whitelist ip, blacklist ip, custom command on detect, and other.
- Make a system that understand when a player spam an ip. Example: "Join in mc server it (with space)" or "Mc|server|it" or "Mc-server-it" or also "Mc0 servername0 it". So the plugin will understand that the player tried to advertise.
- Make ALL messages custom in the config.
- Make a system that block the visibility of the messages when a player try to advertise. Example: "Join in server.it", so the plugin report this to all staff and save it on the file in the folder, all this, while all player can't see the message of the player that tried to advertise.
- System that detect words also in books, signs, chat and under commands (config as Commandtocheck: (list of commands)).
- System that execute a custom commands (also multiple commands) when a player advertise, from config.
- Feature that replace words with other words, like this example:
   #OriginalWord - Word to replace
   Hello - Hi
- Add a singular check for AntiSpam, AntiCommandSpam and AntiCaps with custom commands for each one, like this example:
 #AntiSpam
 EnableCheck: true
 SecondsFOrEachMessages: 3
 CommandsOnPunish:
  - Mute %player Spam is not allowed!
  - Broadcast %player was muted for spam!
And same thig with other 2 checks.

> In this mode this plugin has not false positive, ad hi, i'm ...player .. (points go detect as advertise), and it will be the best.

**Comment by KasperFranz** [comment]
> - Move this project on spigot.

I am not sure what you mean? it is already on spigot - follow the link on the readme.

> - Make a better system of detecting IPs (actual is very bad). So this new detect system will check EVERY words wrote in a text of a player. For Example: "Hi, join in Mc.servername.it!", your plugin will search if this ip works, and if work, the plugin will stop the text from its send. If the text is: "Hi, Join in Mc.serverthatnotexist.it" the plugin will see that this ip is not real and it doesn't stop the send of the text.

The ip checker is already really powerfull, if you have any issues with the current ip checker please create a new issue with what you posted.
If this is to actually verify that there is a minecraft server at the IP, I am note sure that is the best idea - why should it check the ip?

> - Make a new config: whitelist ip, blacklist ip, custom command on detect, and other.

We already make you whitelist certain domains.

In which case would you blacklist a  ip address which already should be blocked because it is in the whitelist?

> - Make a system that understand when a player spam an ip. Example: "Join in mc server it (with space)" or "Mc|server|it" or "Mc-server-it" or also "Mc0 servername0 it". So the plugin will understand that the player tried to advertise.

That is already in place - fell free to make a new issue with which cases isn't found.

> - Make ALL messages custom in the config.

That is on the roadMap to do at some point, as stated on the README we had multiple different languages, if you don't like a transaltion please state so.

> - Make a system that block the visibility of the messages when a player try to advertise. Example: "Join in server.it", so the plugin report this to all staff and save it on the file in the folder, all this, while all player can't see the message of the player that tried to advertise.

That is already in place, if this does not work create an issue.

> -  System that detect words also in books, signs, chat and under commands (config as Commandtocheck: (list of commands)).

I am not sure what you mean, if you mean Detected-Comman then we already have than in the config.

> - System that execute a custom commands (also multiple commands) when a player advertise, from config.
    Feature that replace words with other words, like this example:
    #OriginalWord - Word to replace
    Hello - Hi

This is made in #81

>    Add a singular check for AntiSpam, AntiCommandSpam and AntiCaps with custom commands for each one, like this example:
    #AntiSpam
    EnableCheck: true
    SecondsFOrEachMessages: 3
    CommandsOnPunish:
        Mute %player Spam is not allowed!
        Broadcast %player was muted for spam!
        And same thig with other 2 checks.

Why have it as one?

**Comment by zBoringDrop** [comment]
I Think that you should redevelop the plugin with these suggestion, because your plug-in has lots of false positive .-.

**Comment by KasperFranz** [comment]
Saying I should redevelop it doesn't really help anything - 

What is that a comment to? 

please use markdown to show which comment you are refering to.

**Comment by zBoringDrop** [comment]
No No, i only said that you should develop a plugin with my suggestions, to see if the plugin created is better than yours

**Comment by KasperFranz** [comment]
The code fi by how project has not really been touched in 2 years, I am no longer in bukkit/spigot so it is only of intesrt I have came back to it

**Comment by zBoringDrop** [comment]
So none for my suggestion?

**Comment by KasperFranz** [comment]
I have added comments on all of them, and most are waiting on more information from you

**Comment by zBoringDrop** [comment]
1) Can you resume your plugin on spigot and not on bukkit? It will have more followers
2) IpChecker have lots of bugs, this because this is not an efficent blocking system.
3 \\
4 \\
5 \\
6 \\
7 \\
8 \\
9) I mean that you can create a system that detect an ip wrote on books and signs.
10)  Because if owners use an Anticheat plugin (that already block spam) you can disable spam protection from here.

**Comment by KasperFranz** [comment]
1. The code is the same, as described in the last issue https://github.com/antiAD/AntiAd/issues/77#issuecomment-370253113 there was a prerelease.

2. I need something precise.

3. you can already do that.

4.   already possible

**Comment by zBoringDrop** [comment]
2, most important
A NEW ip blocker
I spam an ip, so AntiAd will check if the spammed ip exist, if yes, it will be execute the command.
If not, none will happen. Only this.

**Comment by KasperFranz** [comment]
2. as said examples and make a issue on it.

**Comment by zBoringDrop** [comment]
sorry, how i can use the whitelisted ips?
Can you send me an example of the format by adding 3 random ip?

**Comment by KasperFranz** [comment]
use the command `/antiad add`

**Comment by zBoringDrop** [comment]
Mhm, /reload and command detection list doesn't work :\

**Comment by KasperFranz** [comment]
I need more info - link a config and log

**Comment by zBoringDrop** [comment]
None, 0 errors and default config.
It works only if i Restart all server

**Comment by KasperFranz** [comment]
share a config and the server log, so ican see what the problem is.
